packages/babel-core/src/transformation/normalize-file.js

```diff
@@ -65,14 +66,18 @@ export default function normalizeFile(
       const lastComment = extractComments(EXTERNAL_SOURCEMAP_REGEX, ast);
       if (typeof options.filename === "string" && lastComment) {
         try {
-          const match = EXTERNAL_SOURCEMAP_REGEX.exec(lastComment);
-          if (!match) throw new Error("Invalid source map comment format.");
-          inputMap = convertSourceMap.fromJSON(
-            fs.readFileSync(
-              path.resolve(path.dirname(options.filename), match[1]),
-              "utf8",
-            ),
+          // when `lastComment` is non-null, EXTERNAL_SOURCEMAP_REGEX must have matches
+          const match: [string, string] = (EXTERNAL_SOURCEMAP_REGEX.exec(
+            lastComment,
+          ): any);
```

Comment:
Is this guaranteed to never return `null`?

Reply:
Yes because the capturing group `([^\s'"`]+)` has a `+` quantifier, if the whole regex passed, the capturing group must not be `null`.